tests/Unit/Numerical/Spectral/Test_DefiniteIntegral.cpp

```diff
@@ -0,0 +1,95 @@
+  const DataVector& x = Basis::lgl::collocation_points(num_pts_in_x);
+  DataVector f(num_pts_in_x);
+  for (size_t a = 0; a < num_pts_in_x; ++a) {
+    for (size_t s = 0; s < f.size(); ++s) {
+      f[s] = pow(x[s], a);
+    }
+    if (0 == a % 2) {
+      CHECK(2.0 / (a + 1.0) == approx(definite_integral(f, index_1D)));
+    } else {
+      CHECK(0.0 == approx(definite_integral(f, index_1D)));
+    }
+  }
+}
+
+void test_definite_integral_2D(const Index<2>& index_2D) {
+  Approx approx = Approx::custom().epsilon(1e-5);
+  Mesh<2> extents(index_2D);
+  const size_t num_pts_in_x = index_2D[0];
+  const size_t num_pts_in_y = index_2D[1];
+  const DataVector& x0 = Basis::lgl::collocation_points(num_pts_in_x);
+  const DataVector& x1 = Basis::lgl::collocation_points(num_pts_in_y);
+  DataVector f(extents.product());
+  for (size_t a = 0; a < num_pts_in_x; ++a) {
+    for (size_t b = 0; b < num_pts_in_y; ++b) {
+      for (IndexIterator<2> I(index_2D); I; ++I) {
+        f[I.offset()] = pow(x0[I()[0]], a) * pow(x1[I()[1]], b);
+      }
+      if (0 == a % 2 and 0 == b % 2) {
+        CHECK(4.0 / ((a + 1.0) * (b + 1.0)) ==
+              approx(definite_integral(f, index_2D)));
+      } else {
+        CHECK(0.0 == approx(definite_integral(f, index_2D)));
+      }
+    }
+  }
+}
+
+void test_definite_integral_3D(const Index<3>& index_3D) {
+  Approx approx = Approx::custom().epsilon(1e-5);
+  Mesh<3> extents(index_3D);
+  const size_t num_pts_in_x = index_3D[0];
+  const size_t num_pts_in_y = index_3D[1];
+  const size_t num_pts_in_z = index_3D[2];
+  const DataVector& x0 = Basis::lgl::collocation_points(num_pts_in_x);
+  const DataVector& x1 = Basis::lgl::collocation_points(num_pts_in_y);
+  const DataVector& x2 = Basis::lgl::collocation_points(num_pts_in_z);
+  DataVector f(extents.product());
+  for (size_t a = num_pts_in_x / 2; a < num_pts_in_x; ++a) {
+    for (size_t b = num_pts_in_y / 2; b < num_pts_in_y; ++b) {
+      for (size_t c = num_pts_in_z / 2; c < num_pts_in_z; ++c) {
+        for (IndexIterator<3> I(index_3D); I; ++I) {
+          f[I.offset()] =
+              pow(x0[I()[0]], a) * pow(x1[I()[1]], b) * pow(x2[I()[2]], c);
+        }
+        if (0 == a % 2 and 0 == b % 2 and 0 == c % 2) {
+          CHECK(8.0 / ((a + 1.0) * (b + 1.0) * (c + 1.0)) ==
+                approx(definite_integral(f, index_3D)));
+        } else {
+          CHECK(0.0 == approx(definite_integral(f, index_3D)));
+        }
+      }
+    }
+  }
+}
+
+TEST_CASE("Unit.Functors.DefiniteIntegral", "[Functors][Unit]") {
+  const size_t start_points = 3;
+  const size_t end_points = 4;
+  for (size_t n0 = start_points; n0 < end_points; ++n0) {
+    test_definite_integral_1D(Index<1>(n0));
+    for (size_t n1 = start_points; n1 < end_points; ++n1) {
+      test_definite_integral_2D(Index<2>(n0, n1));
+      for (size_t n2 = start_points; n2 < end_points; ++n2) {
+        test_definite_integral_3D(Index<3>(n0, n1, n2));
+      }
```

Comment:
- rename start_points -> min_extent  (or maybe just put the numbers in the loops?)
- rename end_points -> max_extent
- change < to <=
- change min_extent to 2
- split into separate loops for 1, 2, 3 dimensions
- in 1D set max_extent = Basis::lgl::maximum_number_of_pts
- in 2D try to keep the max if test doesn't take too long
- in 3D set max_extent so test doesn't take too long